lang/java/reef-bridge-java/src/main/java/org/apache/reef/javabridge/generic/JobDriver.java

```diff
@@ -118,13 +121,14 @@
    */
   private final LoggingScopeFactory loggingScopeFactory;
   private final Set<String> definedRuntimes;
+  private final DateFormat dateFormat = new SimpleDateFormat("yyyy/MM/dd HH:mm:ss");
 
   private BridgeHandlerManager handlerManager = null;
   private boolean isRestarted = false;
   // We are holding on to following on bridge side.
   // Need to add references here so that GC does not collect them.
-  private final HashMap<String, AllocatedEvaluatorBridge> allocatedEvaluatorBridges =
-      new HashMap<>();
+  private final ConcurrentHashMap<String, AllocatedEvaluatorBridge> allocatedEvaluatorBridges =
```

Comment:
Not sure this has the semantics we'd need here. `ConcurrentHashMap` allows retrieval and update functions to be happening in parallel, which can introduce hard to debug cases. Would adding small `synchronized` blocks around the operations on the maps be too much of perf penalty? If yes, we need to think this through carefully.

Reply:
I guess that is what ConcurrentHashMap suppose to be to make operation atomic. 

synchronized block on the map should work as well as we don't lock it very long.